src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc

```diff
@@ -0,0 +1,136 @@
+#include "org_bblfsh_client_v2_libuast_Libuast.h"
+#include "jni_handle.h"
+#include "jni_utils.h"
+#include "org_bblfsh_client_v2_Context.h"
+
+#include "libuast.h"
+#include "libuast.hpp"
+
+JavaVM *jvm;  // FIXME(bzz): double-check and document
```

Comment:
We should almost certainly put this symbol into a namespace unless for some reason we can't. I'm also concerned about initialization, since there is a link-time dependency with `jni_utils.cc` and I can't currently tell for sure that we actually get the correct order. 

I second your TODO, and would add that we should document not only what it is, but how and where it is populated. If you haven't already considered it, I would think about providing a factory function that does the initialization on demand and returns a pointer, which would allow us to anonymously namespace the symbol and protect us from dependency order oddities up to concurrency.

```c
JavaVM* getJvm() {
    static JavaVM* instance = nullptr;
    static bool initialized = false;

    if (!initialized) {
        instance = new JavaVM;  // or whatever
        initialized = true;
    }
    return instance;
}
```

or words to that effect. (Or in an anonymous namespace at file scope, if you dislike statics)

Reply:
This is a very valuable feedback - thank you! That's a piece of code left from previous implementation, it's now initialized while loading the JNI (guaranteed to be once) and if you do not mind, I would prefer dealing with this in a subsequent PRs, as for now it's "just the same as before" and should not be treated as a change. 

I'll add a TODO with the link to this thread but please, let me know if you think it's crucial to have some immediate improvements (e.g. namespace)

Reply:
This can definitely wait.